node/src/components/block_validator.rs

```diff
@@ -320,13 +300,16 @@ where
                     if state.missing_deploys.remove(&dt_hash) {
                         // If the deploy is of the wrong type or would be invalid for this block,
                         // notify everyone still waiting on it that all is lost.
-                        if deploy_type.is_transfer() != dt_hash.is_transfer() {
-                            info!(block = ?key, %dt_hash, ?deploy_type, "wrong deploy type");
-                            invalid.push(key.clone());
-                        } else if let Err(err) =
-                            state.appendable_block.add(dt_hash.into(), &*deploy_type)
-                        {
-                            info!(block = ?key, %dt_hash, ?deploy_type, ?err, "block invalid");
+                        let add_result = match dt_hash {
+                            DeployOrTransferHash::Deploy(hash) => {
+                                state.appendable_block.add_deploy(hash, &*deploy_info)
+                            }
+                            DeployOrTransferHash::Transfer(hash) => {
+                                state.appendable_block.add_transfer(hash, &*deploy_info)
+                            }
+                        };
```

Comment:
Now that `DeployInfo` doesn't contain the information anymore whether this _actually_ is a transfer or not, someone could send us a block again with the two mixed up, and we wouldn't notice. We probably need to check in `fetch_deploy` whether it's as expected. If it isn't, the block is invalid.